gossipd/gossip.c

```diff
@@ -180,6 +347,41 @@ static bool send_peer_with_fds(struct peer *peer, const u8 *msg);
 static void wake_pkt_out(struct peer *peer);
 static bool try_reach_peer(struct daemon *daemon, const struct pubkey *id);
 
+
+
+
+// called when we want to connect to TOR SOCKS5
+static struct io_plan *io_tor_connect(struct io_conn *conn, struct reaching *reach)
+{
+	static struct addrinfo *ai_tor;
+	static char *port_addr;
+
```

Comment:
You should probably do something like `struct reaching_socks *reach_tor = tal(reach, struct reaching_socks);` here.